modules/javafx.controls/src/test/java/test/javafx/scene/control/ListViewTest.java

```diff
@@ -1714,6 +1714,11 @@ public RT22599_DataType(int id, String text) {
             if (obj == null) return false;
             return id == ((RT22599_DataType)obj).id;
```

Comment:
This does look dubious. Since this was written for a specific unit test, it might have been intentional to have a "wrong" equals method, but if so, a comment should have been added to that effect. In any case, fixing `equals` is out of scope for this PR. And given what `equals` does, the proposed `hashCode` method is correct.